### ISSUE TITLE
[VoiceOver] Published Date Picker issues

### ISSUE THREAD

**Issue by shiki** [wordpress-mobile/WordPress-iOS#12959]
<img src="https://user-images.githubusercontent.com/198826/68971505-e23f8280-07a6-11ea-9dbd-ac4f45eef6f9.png" width="320">

This is a new-ish component and generally needs some accessibility work. 

- [ ] The current date can have a single `accessibilityLabel` with an updated `accessibilityValue`
- [ ] The _S_, _M_, _T_, and other labels do not need to be accessible. Instead, the days should have the labels “Monday. March 14”
- [x] #13056 The picker cannot be dismissed using the VoiceOver global escape
- [x] #13056 The Close button has no accessibility label
- [ ] When swiping between months, VoiceOver says something like “Page 800 of 1000”. This doesn't seem very useful.
- [ ] Left and Right buttons do not have labels
- [ ] The day buttons can use the `.button` and `.selected` traits

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
